From 2767088c4623fd9fffb9933f62161206eeda62ba Mon Sep 17 00:00:00 2001
From: "Spencer E. Olson" <olsonse@umich.edu>
Date: Tue, 31 May 2022 05:53:41 -0600
Subject: [PATCH 6/8] add commandline option to show build version

Signed-off-by: Spencer E. Olson <olsonse@umich.edu>
---
 src/aklog-kafs.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Index: kafs-client/src/aklog-kafs.c
===================================================================
--- kafs-client.orig/src/aklog-kafs.c	2022-12-10 23:39:14.840847958 +0000
+++ kafs-client/src/aklog-kafs.c	2022-12-10 23:39:14.828847380 +0000
@@ -354,6 +354,7 @@
 		"         uid-session\n"
 		"       Otherwise, a session keyring will be used first if found before \n"
 		"       automatically switching to the uid-session keyring.\n"
+		" -V    Show version and exit\n"
 	);
 	exit(1);
 }
@@ -378,7 +379,7 @@
 
 	dest_keyring = 0;
 
-	while ((opt = getopt(argc, argv, "hvk:")) != -1) {
+	while ((opt = getopt(argc, argv, "hvVk:")) != -1) {
 		switch (opt) {
 		case 'h':
 			opt_help = true;
@@ -394,6 +395,9 @@
 		case 'v':
 			++opt_verbose;
 			break;
+		case 'V':
+			printf("kAFS client: %s\n", VERSION);
+			exit(0);
 		default:
 			display_usage();
 		}
