From 04bbe4091f3383390aa62b0f1de5f0f2a2f131e3 Mon Sep 17 00:00:00 2001
From: "Spencer E. Olson" <olsonse@umich.edu>
Date: Sat, 28 May 2022 15:22:43 -0600
Subject: [PATCH 3/8] remove -d (debug) option and improve -v (verbose) option

Signed-off-by: Spencer E. Olson <olsonse@umich.edu>
---
 src/aklog-kafs.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

Index: kafs-client/src/aklog-kafs.c
===================================================================
--- kafs-client.orig/src/aklog-kafs.c	2022-12-10 23:10:31.077841066 +0000
+++ kafs-client/src/aklog-kafs.c	2022-12-10 23:14:44.758033240 +0000
@@ -35,7 +35,6 @@
 #include <linux/if_alg.h>
 
 // command line switches
-bool opt_debug   = false;
 bool opt_help    = false;
 bool opt_verbose = false;
 
@@ -343,7 +342,7 @@
  */
 void display_usage ()
 {
-	fprintf(stderr, "Usage: aklog-kafs [-dhv] [<cell> [<realm>]]\n");
+	fprintf(stderr, "Usage: aklog-kafs [-hv] [<cell> [<realm>]]\n");
 	exit(1);
 }
 
@@ -364,12 +363,8 @@
 	krb5_ccache cc;
 	krb5_creds search_cred, *creds;
 
-	while ((opt = getopt(argc, argv, "dhv")) != -1) {
+	while ((opt = getopt(argc, argv, "hv")) != -1) {
 		switch (opt) {
-		case 'd':
-			opt_debug = true;
-			opt_verbose = true;
-			break;
 		case 'h':
 			opt_help = true;
 			break;
@@ -408,6 +403,9 @@
 		for (p = realm; *p; p++)
 			*p = toupper(*p);
 	}
+	if (opt_verbose) {
+ 		printf("Realm: %s\n", realm);
+ 	}
 
 	for (p = cell; *p; p++)
 		*p = tolower(*p);
@@ -466,6 +464,9 @@
 
 	ret = add_key("rxrpc", desc, payload, plen, KEY_SPEC_SESSION_KEYRING);
 	OSERROR(ret, "add_key");
+	if (opt_verbose) {
+		printf("successfully added key: %d to session keyring\n", ret);
+	}
 
 	if (cell_scratch) {
 		free(cell_scratch);
