From 397420ff8e55c56006e68c11b663331e6240c1c2 Mon Sep 17 00:00:00 2001
From: "Spencer E. Olson" <olsonse@umich.edu>
Date: Tue, 31 May 2022 05:10:28 -0600
Subject: [PATCH 4/8] enable -v option to incrementally increase verbosity

Signed-off-by: Spencer E. Olson <olsonse@umich.edu>
---
 src/aklog-kafs.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

Index: kafs-client/src/aklog-kafs.c
===================================================================
--- kafs-client.orig/src/aklog-kafs.c	2022-12-10 23:15:58.245565556 +0000
+++ kafs-client/src/aklog-kafs.c	2022-12-10 23:19:49.588700553 +0000
@@ -36,7 +36,7 @@
 
 // command line switches
 bool opt_help    = false;
-bool opt_verbose = false;
+int opt_verbose = 0;
 
 struct rxrpc_key_sec2_v1 {
         uint32_t        kver;                   /* key payload interface version */
@@ -342,7 +342,14 @@
  */
 void display_usage ()
 {
-	fprintf(stderr, "Usage: aklog-kafs [-hv] [<cell> [<realm>]]\n");
+	fprintf(stderr,
+		"Usage: \n"
+		" aklog-kafs [OPTIONS] [<cell> [<realm>]]\n"
+		"\n"
+		"Options:\n"
+		" -h    display this help and exit\n"
+		" -v    increase verbosity with each instance of this argument\n"
+	);
 	exit(1);
 }
 
@@ -369,7 +376,7 @@
 			opt_help = true;
 			break;
 		case 'v':
-			opt_verbose = true;
+			++opt_verbose;
 			break;
 		default:
 			display_usage();
@@ -449,9 +456,11 @@
 		exit(1);
 	}
 
-	printf("plen=%zu tklen=%u rk=%zu\n",
-	       plen, creds->ticket.length, sizeof(*payload));
-
+	if (opt_verbose >= 2) {
+		printf("plen=%zu tklen=%u rk=%zu\n",
+			plen, creds->ticket.length, sizeof(*payload));
+	}
+    
 	/* use version 1 of the key data interface */
 	payload->kver           = 1;
 	payload->security_index = 2;
