From ae1cd01d1d416a5de70a7a077b6a44f50d89a350 Mon Sep 17 00:00:00 2001
From: "Spencer E. Olson" <olsonse@umich.edu>
Date: Tue, 31 May 2022 05:56:25 -0600
Subject: [PATCH 7/8] make display_usage use proper exit status

Signed-off-by: Spencer E. Olson <olsonse@umich.edu>
---
 src/aklog-kafs.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

Index: kafs-client/src/aklog-kafs.c
===================================================================
--- kafs-client.orig/src/aklog-kafs.c	2022-12-10 23:48:33.843784846 +0000
+++ kafs-client/src/aklog-kafs.c	2022-12-10 23:50:05.780215099 +0000
@@ -340,7 +340,7 @@
 /*
  * Display a short usage message and exit
  */
-void display_usage ()
+void display_usage (int exit_status)
 {
 	fprintf(stderr,
 		"Usage: \n"
@@ -356,7 +356,7 @@
 		"       automatically switching to the uid-session keyring.\n"
 		" -V    Show version and exit\n"
 	);
-	exit(1);
+	exit(exit_status);
 }
 
 /*
@@ -382,7 +382,7 @@
 	while ((opt = getopt(argc, argv, "hvVk:")) != -1) {
 		switch (opt) {
 		case 'h':
-			opt_help = true;
+			display_usage(EXIT_SUCCESS);
 			break;
 		case 'k':
 			if (strcmp(optarg, "session") == 0)
@@ -390,7 +390,7 @@
 			else if (strcmp(optarg, "uid-session") == 0)
 				dest_keyring = KEY_SPEC_USER_SESSION_KEYRING;
 			else
-				display_usage();
+				display_usage(EXIT_FAILURE);
 			break;
 		case 'v':
 			++opt_verbose;
@@ -399,16 +399,16 @@
 			printf("kAFS client: %s\n", VERSION);
 			exit(0);
 		default:
-			display_usage();
+			display_usage(EXIT_SUCCESS);
 		}
 	}
 
 	if (argc - optind > 2) {
 		fprintf(stderr, "ERROR: too many arguments\n");
-		display_usage();
+		display_usage(EXIT_FAILURE);
 	}
 	if (strcmp(argv[optind], "--help") == 0) {
-		display_usage();
+		display_usage(EXIT_SUCCESS);
 	}
 
 	if ((argc - optind) <= 0) {
